content/articles/mern-isp-management-system/index.md

```diff
@@ -0,0 +1,611 @@
+
+## Table of Contents
+
+- [Setting Up the Backend](#chapter-1)
+
+- [Setting Up the Frontend and Connecting it to the Backend](#chapter-2)
+
+- [Conclusion](#chapter-3)
+
+## Prerequisites
+
+1. An understanding of JavaScript.
+2. Basic knowledge of React and Node JS.
+3. Basic knowledge of MongoDB.
+4. You should also have [Node](https://nodejs.org/en/download/) installed on your machine and have a tool for testing API endpoints such as [Postman](https://www.postman.com/downloads/).
+
+## Setting Up the Backend<a name="chapter-1"></a>
+
+We will handle all the backend functionality of our app using a REST API created using Node, Express, and MongoDB.
+
+### Setting up Development Environment
+
+Let us first set up our development environment.
+Create a new folder.
+
+`mkdir api`
+
+Enter the directory
+
+`cd api`
+
+Inside this folder, run `npm init--y `to get a starter package.json which will contain all the dependencies we will need to run the app.
+Our package.json file gets updated when we install various dependencies.
+We need to install express to set up a node server. 
+`npm install express`
+Then we install MongoDB, which will enable us to connect to the database, and mongoose, which we will use for defining models.
+
+`npm install mongodb`
+`npm install mongoose`
+
+We will be sharing resources from the server to the client; therefore, we install cors to handle cross-origin resource sharing.
+
+`npm install cors`
+
+We will install nodemon to run our server automatically when we make changes.
+
+`npm install nodemon --save-dev`
+
+We have to change the scripts in package.json to 
+`"devStart": "nodemon index.js"`
+We will also install dotenv to handle our variables from a .env file.
+
+`npm install dotenv --save-dev`
+
+Note that nodemon and dotenv are in our development dependencies.
+
+### Setting up the Server
+
+Create a .env file with the following code.
+
+```
+DATABASE_URL = mongodb://localhost/customers
+PORT = 5000
+```
+Let us create index.js and add the following code.
+
+```JavaScript
+require('dotenv').config()
+const cors = require('cors')
+const express = require('express')
+const app = express()
+const mongoose = require('mongoose')
+const PORT = process.env.PORT || 5010;
+
+app.use(express.json())
+app.use(cors())
+
+app.listen(PORT, () => console.log(`Server Started on port ${PORT}`))
+
+```
+
+First, we import all the libraries we installed.
+We define the app variable as an instance of express, which we will use to configure our server.
+
+app. use() allows us to use the different libraries needed, including `app. use(express.json())`, which is a middleware function that parses incoming JSON requests and puts the parsed data in req. body.
+app. listen()listens to the connections on the specified port and executes the callback function we pass.
+process.env allows us to access environment variables 
```

Comment:
Convert these points to bulleted points. And, enclose them within ` `` `

Reply:
I see that these commands might give compiler errors if used as such (Note the extra space in the commands)